src/parsers/manifest/dash/indexes/base.ts

```diff
@@ -165,27 +167,41 @@ export default class BaseRepresentationIndex implements IRepresentationIndex {
                                     representationId,
                                     representationBitrate);
 
-    // TODO If indexRange is either undefined or behind the initialization segment
+    // TODO If indexRange is behind the initialization segment
     // the following logic will not work.
-    // However taking the nth first bytes like `dash.js` does (where n = 1500) is
-    // not straightforward as we would need to clean-up the segment after that.
-    // The following logic corresponds to 100% of tested cases, so good enough for
-    // now.
-    const range : [number, number] | undefined =
-      index.initialization !== undefined ? index.initialization.range :
-      index.indexRange !== undefined ? [0, index.indexRange[0] - 1] :
-                                       undefined;
+
+    const isMP4 = mimeType !== undefined &&
+                  /\/mp4$/.exec(mimeType) !== null;
+
+    const initialization = (() => {
+      if (index.initialization === undefined && !isMP4) {
+        return null; // no init segment in content
+      }
+
+      const range = index.initialization?.range ??
+        (index.indexRange !== undefined ? [0, index.indexRange[0] - 1] :
+                                          undefined);
+      return { mediaURL, range }; // there may be an init segment
+    })();
 
     this._index = { indexRange: index.indexRange,
                     indexTimeOffset,
-                    initialization: { mediaURL, range },
+                    initialization,
                     mediaURL: createIndexURL(representationBaseURL,
                                              index.media,
                                              representationId,
                                              representationBitrate),
                     startNumber: index.startNumber,
                     timeline: index.timeline,
                     timescale };
+
+    if (initialization === null &&
+        index.timeline.length === 0 &&
+        index.indexRange === undefined) {
```

Comment:
Weird to re-check later in the same function. Why not including both in the same `if`?